### ISSUE TITLE
Maps got frozen after parent render!  For instance, Material UI theme change.

### ISSUE THREAD

**Issue by dennisat** [PaulLeCam/react-leaflet#888]
## Bug report

- [x] All peer dependencies are installed: React, ReactDOM and Leaflet.
- [x] Using a supported version of React and ReactDOM (v17.0.0 minimum).
- [x] Using the supported version of Leaflet (v1.7.1 minimum) and its corresponding CSS file is loaded.
- [x] Using the [latest v3 version of React-Leaflet](https://github.com/PaulLeCam/react-leaflet/releases).
- [x] The issue has not already been reported.
- [x] Make sure you have followed the [quick start guide](http://leafletjs.com/examples/quick-start.html) for Leaflet.
- [x] Make sure you have fully read the [documentation](https://react-leaflet.js.org/docs/start-introduction) and that you understand the [limitations](https://react-leaflet.js.org/docs/start-introduction#limitations).

### Expected behavior

When the Material UI Theme changes, the map should still work as expected.

### Actual behavior

When the Material UI theme changes (from dart to white and vice versa), the map becomes broken.

- The map is not draggable, except if you drag it from the Circle are (see screenshot)
- Sometimes, the map has some horizontal gaps of one pixel.

![image](https://user-images.githubusercontent.com/26453466/119477670-705ad880-bd4f-11eb-8086-1620e009a99f.png)

### Steps to reproduce

Change MUI theme. This triggers a render, the map is rendered but it's broken.

### Investigation

It looks like some event handlers are not bound currently. Although there is no flick.

**There is another way to reproduce this bug**

In this application, the Map is under a 2 level (Material UI) Tab component.
We can reproduce this problem in the following way:

- We have a <form> with Material UI Tab Components and there are our Maps (many maps).
- On the 1st Tab Component Select the "Maps" tab
- On the 2nd _nested_ Maps Tab Component Select the "Area" Tab. This 2nd Tab Component has 4 tabs with Maps on each one.
- The "Area" Map is shown
- _Now let's go back on 1st Tab Component to make a change on another tab of the 1st Tab Component_
- On 1st Tab Component Select the "Notes" tab
- This tab has TextArea Component
- Enter something in the TextArea
- **KEY POINT!** as we change the TextArea's content, a render is taking place on the whole form. **This render is breaking the maps for some reason!** The map is unmounted at this time since belongs on another tab, and in precise a nested tab.
- _Now let's go back on the map_
- On the 1st Tab Component Select the "Maps" tab
- On the 2nd _Nested_ Maps Tab Component Select the "Area" Tab
- The "Area" Map is shown
- **Bug:** Now if you try to drag the map is not working. Also, some strange horizontal gaps are on the map.

**Workaround**

For this problem with the Tabs, if a render the map after 1sec, it works! But the map breaks on MUI Theme change.

```
  const [bugfixShow, setBugfixShow] = useState(false);
  useEffect(() => {
    setTimeout(() => setBugfixShow(true), 1000);
  }, []);
  // ...
  if (!bugfixShow) return <div/>;
  // ...
  return <MapContainer...
```

**To sum up**

There is an issue when a parent component triggers a render. This is clean, especially on the MUI Theme change.

**Comment by achepukov** [comment]
I also noticed something similar, is there a similar link to codepen for 3.x https://codepen.io/PaulLeCam/pen/gzVmGw, so we can extract the case?

**Comment by achepukov** [comment]
@PaulLeCam I feel that the root case of this issue is that create function is called even on update, failed test case has linked

**Comment by achepukov** [comment]
Hi @PaulLeCam , [I updated the PR, with possible fix](https://github.com/achepukov/react-leaflet/pull/1/commits/5d26a9ea89ebe5a848b23edb81b60a4981dc8864) - memoize the created element on mount using the `useMemo` hook. This makes the test green, and theoretically is a correct fix, however, I'm not sure it fits the general architecture. Could you please review?

**Comment by PaulLeCam** [comment]
Please provide a codepen reproducing the issue without a third-party library.
